src/lib/all_type_variant.hpp

```diff
@@ -26,23 +26,26 @@ namespace detail {
 #define EXPAND_TO_HANA_TYPE(s, data, elem) boost::hana::type_c<elem>
 
 // clang-format off
-#define COLUMN_TYPES                                  (int32_t) (int64_t) (float)  (double)  (std::string)    // NOLINT
-static constexpr auto type_strings = hana::make_tuple("int",    "long",   "float", "double", "string"     );  // NOLINT
+#define DATA_TYPES                     (int32_t) (int64_t) (float)  (double)  (std::string)  // NOLINT
+enum class DataType : uint8_t { Null, Int,      Long,     Float,   Double,   String };     // NOLINT
 // clang-format on
 
+static constexpr auto data_type_enum_values =
+    hana::make_tuple(DataType::Int, DataType::Long, DataType::Float, DataType::Double, DataType::String);
```

Comment:
Is there any way we can avoid having to list those a third time?

Reply:
Another macro maybe?